test/controllers/whototell.controller.test.ts

```diff
@@ -0,0 +1,59 @@
+import 'reflect-metadata'
+
+import { assert } from 'chai'
+import { BAD_REQUEST, MOVED_TEMPORARILY, OK } from 'http-status-codes'
+import request from 'supertest'
+import { deepEqual, instance, mock, when } from 'ts-mockito'
+
+import { createApp } from 'app/application.factory'
+import 'app/controllers/whototell.controller'
+import ValidationErrors from 'app/models/validationErrors'
+import { WHO_TO_TELL_URI } from 'app/paths'
+import formSchema from 'app/schemas/whototell.schema'
+import FormValidator from 'app/utils/formValidator.util'
+
+describe.only('FormController', () => {
+  describe('GET - ensure that page loads correctly', () => {
+    it('should render the form page', async () => {
+      const app = createApp()
+
+      const res = await request(app).get(WHO_TO_TELL_URI).expect(OK)
+
+      assert.include(res.text, 'Who to tell about the company closing')
+    })
+  })
+
+  describe('POST - ensure form submission is handled correctly', () => {
+    it('should redirect successfully if validator returns no errors', async () => {
+      const testObject = {test: 'data'}
+
+      const mockedFormValidator = mock(FormValidator)
+      when(mockedFormValidator.validate(deepEqual(testObject), formSchema)).thenReturn(null)
+
+      const app = createApp(container => {
+        container.rebind(FormValidator).toConstantValue(instance(mockedFormValidator))
+      })
+
+      const res = await request(app).post(WHO_TO_TELL_URI).send(testObject).expect(MOVED_TEMPORARILY)
+
+      assert.include(res.text, 'Found. Redirecting to /close-a-company/search-company')
+    })
+  })
+
+  it('should render view with errors displayed if validator returns errors', async () => {
+    const testData = {test: 'data'}
+    const mockError: ValidationErrors = {
+      someTextField: 'Im a validation error'
```

Comment:
Hmmm, the problem with this is that the error will be shown at the top of the page in the summary, but it won't be shown inline above the checkbox. We have two options:
1. Assert that there are two instances of this error message (once in the summary, once inline)
2. or, introduce a HTML element assertion library so that we can target specific elements and verify their content i.e. 
```
assert.isTrue(html.hasText('#confirmation.govuk-error', 'Im a validation error'))
```

Option 1 is easier, but Option 2 provides better coverage. Maybe we can proceed with Option 1 for now, and introduce Option 2 later? What do you think?

Reply:
Good point, I like Option 2 but it should probably be implemented later. For now I will implement Option 1 and we can look at it again soon.